### PR TITLE
Apply default configuration if parameter is not present.

### PR DESCRIPTION
**Pull request by sbussweiler**
Signed-off-by: Stefan Bussweiler <s.bussweiler@external.telekom.de>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java [hunk 1]
```diff
@@ -601,7 +600,10 @@ public Void call() throws Exception {
     }
 
     private void initializeHandler(final Thing thing) {
-        if (isInitializable(thing)) {
+        ThingType thingType = thingTypeRegistry.getThingType(thing.getThingTypeUID());
+        ThingFactoryHelper.applyDefaultConfiguration(thing.getConfiguration(), thingType, configDescriptionRegistry);
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java]: This call will lead to an NPE if no thing type is found for the thing.

**Comment by kaikreuzer** [comment]
Another thought: This PR now changes the configuration of a Thing without persisting it - which means that the persisted config stays as before and is not in sync with what the runtime sees.
Is this something we are fine with? It reminds me of a similar discussion about properties, where at least a binding was not allowed to do changes to properties, since persisting them might not work (e.g. if defined in DSL files).

**Comment by kaikreuzer** [comment]
> Is this something we are fine with?

I answer it myself: Yes, we are, since this is actually already the case - Things from the DSL are already now "enriched" with required configuration values, if they are not present, so this PR does not really change anything fundamental here.

**Comment by kaikreuzer** [comment]
@sbussweiler Can this be due to your change?
https://hudson.eclipse.org/smarthome/job/SmartHomeDistribution-Nightly/lastCompletedBuild/testReport/org.eclipse.smarthome.core.thing.internal/ThingLinkManagerOSGiTest/assert_that_existing_things_are_linked/

I would not really see why, but this PR was the only change in this build...

**Comment by kaikreuzer** [comment]
Looking a bit closer, it indeed seems to be related to your changes:

```
Running org.eclipse.smarthome.core.thing.binding.ThingFactoryTest
05:16:22.675 [main] WARN  o.e.s.c.thing.binding.ThingFactory - Could not parse default value 'invalid' as type 'DECIMAL': null
java.lang.NumberFormatException: null
	at java.math.BigDecimal.<init>(BigDecimal.java:470) ~[na:1.7.0_80]
	at java.math.BigDecimal.<init>(BigDecimal.java:739) ~[na:1.7.0_80]
	at org.eclipse.smarthome.core.thing.internal.ThingFactoryHelper.getDefaultValueAsCorrectType(ThingFactoryHelper.java:157) [org.eclipse.smarthome.core.thing-0.8.0-SNAPSHOT.jar:na]
	at org.eclipse.smarthome.core.thing.internal.ThingFactoryHelper.applyDefaultConfiguration(ThingFactoryHelper.java:189) [org.eclipse.smarthome.core.thing-0.8.0-SNAPSHOT.jar:na]
	at org.eclipse.smarthome.core.thing.binding.ThingFactory.createThing(ThingFactory.java:98) [org.eclipse.smarthome.core.thing-0.8.0-SNAPSHOT.jar:na]
```

**Comment by sbussweiler** [comment]
Fixed by #1493

**Pull request merged**